Warnings on MacOS

```
...
[18/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/cpu/cpu_unified_buffer.cpp.o
  /Users/runner/work/xmipp4-core/xmipp4-core/src/compute/cpu/cpu_unified_buffer.cpp:88:5: warning: expression result unused [-Wunused-value]
      dynamic_cast<cpu_device_queue&>(queue); // Check
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 warning generated.
  [19/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/cpu/cpu_unified_memory_allocator.cpp.o
  /Users/runner/work/xmipp4-core/xmipp4-core/src/compute/cpu/cpu_unified_memory_allocator.cpp:32:5: warning: expression result unused [-Wunused-value]
      dynamic_cast<cpu_device_queue&>(queue);
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 warning generated.
  [20/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/device_buffer.cpp.o
  [21/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/device_manager.cpp.o
  [22/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/host_buffer.cpp.o
  [23/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/unified_buffer.cpp.o
  [24/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/unified_copy.cpp.o
  [25/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/compute/unified_memory_allocator.cpp.o
  [26/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/core_version.cpp.o
  [27/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/interface_catalog.cpp.o
  [28/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/memory/aligned_alloc.cpp.o
  /Users/runner/work/xmipp4-core/xmipp4-core/src/memory/aligned_alloc.cpp:19:16: warning: 'aligned_alloc' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
          return std::aligned_alloc(alignment, size);
                 ^~~~~~~~~~~~~~~~~~
  /Applications/Xcode_15.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/malloc/_malloc.h:65:7: note: 'aligned_alloc' has been marked as being introduced in macOS 10.15 here, but the deployment target is macOS 10.13.0
  void *aligned_alloc(size_t __alignment, size_t __size) __result_use_check __alloc_align(1) __alloc_size(2) _MALLOC_TYPED(malloc_type_aligned_alloc, 2) __OSX_AVAILABLE(10.15) __IOS_AVAILABLE(13.0) __TVOS_AVAILABLE(13.0) __WATCHOS_AVAILABLE(6.0);
        ^
  /Users/runner/work/xmipp4-core/xmipp4-core/src/memory/aligned_alloc.cpp:19:16: note: enclose 'aligned_alloc' in a __builtin_available check to silence this warning
          return std::aligned_alloc(alignment, size);
                 ^~~~~~~~~~~~~~~~~~
  1 warning generated.
  [29/37] Building CXX object CMakeFiles/xmipp4-core.dir/src/plugin_loader.cpp.o
...
```

--- a/include/xmipp4/core/multidimensional/slice.inl
+++ b/include/xmipp4/core/multidimensional/slice.inl
@@ -506,7 +506,7 @@ void sanitize_slice_count(dynamic_slice &slice, std::size_t extent)
             const auto new_count = (extent - start + step - 1) / step;
             slice.set_count(new_count);
         }
-        else if (count > 0 && start + step*(count-1) >= extent)
+        else if (count > 0U && start + step*(count-1) >= extent)
         {
             std::ostringstream oss;
             oss << "Slice count " << count 
@@ -524,7 +524,7 @@ void sanitize_slice_count(dynamic_slice &slice, std::size_t extent)
             const auto new_count = (start + abs_step - 1) / abs_step + 1;
             slice.set_count(new_count);
         }
-        else if (count > 0U && abs_step*(count-1) > start)
+        else if (count > 0U && abs_step*static_cast<std::ptrdiff_t>(count-1) > start)
         {
             std::ostringstream oss;
             oss << "Reversed slice with count " << count 

--- a/src/compute/cpu/cpu_unified_buffer.cpp
+++ b/src/compute/cpu/cpu_unified_buffer.cpp
@@ -85,7 +85,7 @@ const void* cpu_unified_buffer::get_data() const noexcept
 
 void cpu_unified_buffer::record_queue(device_queue& queue)
 {
-    dynamic_cast<cpu_device_queue&>(queue); // Check
+    std::ignore = dynamic_cast<cpu_device_queue&>(queue); // Check
     // No-op
 }
 

--- a/src/compute/cpu/cpu_unified_memory_allocator.cpp
+++ b/src/compute/cpu/cpu_unified_memory_allocator.cpp
@@ -29,7 +29,7 @@ cpu_unified_memory_allocator::create_unified_buffer(std::size_t size,
                                                     std::size_t alignment,
                                                     device_queue &queue )
 {
-    dynamic_cast<cpu_device_queue&>(queue);
+    std::ignore = dynamic_cast<cpu_device_queue&>(queue);
     return create_unified_buffer_impl(size, alignment);
 }
 

--- a/src/memory/aligned_alloc.cpp
+++ b/src/memory/aligned_alloc.cpp
@@ -15,18 +15,26 @@ void* aligned_alloc(std::size_t size, std::size_t alignment) noexcept
 {
     #if XMIPP4_WINDOWS
         return _aligned_malloc(size, alignment);
+    #elif XMIPP4_POSIX
+        void *result;
+        
+        if(posix_memalign(&result, alignment, size) != 0)
+        {
+            result = nullptr;
+        }
+
+        return result;
     #else
         return std::aligned_alloc(alignment, size);
     #endif
-
 }
 
 void aligned_free(void* ptr) noexcept
 {
     #if XMIPP4_WINDOWS
-        return _aligned_free(ptr);
+        _aligned_free(ptr);
     #else
-        return std::free(ptr);
+        std::free(ptr);
     #endif
 }
 

--- a/src/plugin_manager.cpp
+++ b/src/plugin_manager.cpp
@@ -148,6 +148,8 @@ void discover_plugins(const std::string& directory, plugin_manager &manager)
             directory, 
             e.what()
         );
+
+        std::ignore = e; // To avoid compiler warnings with log disabled
     }
 
     for (const auto& entry : iterator) 

